core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -117,6 +117,11 @@ public long getSizeInBytes()
     private void appendDictionary(IntArrayList positions, DictionaryBlock source)
     {
         Block dictionary = source.getDictionary();
+        if (dictionary instanceof RunLengthEncodedBlock rleDictionary) {
+            appendRle(rleDictionary.getValue(), positions.size());
+            return;
+        }
+
```

Comment:
It is possible at the moment but it will be brittle as we don't want blocks like that so the test may stop covering this case at any time e.g. after https://github.com/trinodb/trino/pull/17842 is merged